rfd/0009-viewer-app.md

```diff
@@ -0,0 +1,76 @@
+---
+authors: Mathew (wrm932@gmail.com)
+state: draft
+---
+
+# Required Approvers
+* Engineering @pierrebeaucamp && @jimbishopp && @michellescripts && @mcbattirola
+
+# RFD 0009 - RFDs
+
+## What
+
+Create a directory viewer similar to Github and Google drive. 
+
+
+## Why
+
+A pratical exam that tests the potential applicant on their fullstack software knowledge. 
+
+## Details
+
+Each RFD is stored in a markdown file under
+https://github.com/gravitational/teleport/tree/master/rfd and has a unique
+number.
+
+The directory viewer should be allow the user to navigate a directory structure by clicking on files or folders. A bread crumb navigation should inform them of where they currently are and allow them to navigate back to their previously clicked directory location. The user should also be able to filter out a list of files using a search box. Search/filtering will be done client side. 
```

Comment:
> navigate a directory structure by clicking on files or folders 

I just wanted to note that files should be a no-op == no navigation.

I would love to see some simple wireframes as well - nothing extravagant.

Reply:
Here's my best attempt at a wireframe! Please forgive me...

<img width="609" alt="Screen Shot 2022-06-28 at 11 09 33 PM" src="https://user-images.githubusercontent.com/21994/176364024-10736184-5f38-4f98-bd38-1496dd0d8749.png">

Reply:
looks good 👍.

Please keep in mind that you don't have to report the 'modified' timestamp. You also don't need to provide different icons for different file types. The only distinction we want to see is between a file and a directory.